src/hooks/useTimeRemainingInBatch.ts

```diff
@@ -13,3 +14,70 @@ export function useTimeRemainingInBatch(): number {
 
   return timeRemaining
 }
+
+interface SecondsRemainingResult {
+  secondsRemaining: number
+  checkTime: boolean
+}
+
+const checkIfTime = (seconds: number): SecondsRemainingResult => {
+  const secondsRemaining = getSecondsRemainingInBatch()
+  return {
+    secondsRemaining,
+    checkTime: secondsRemaining < seconds,
+  }
+}
+
+export function useCheckWhenTimeRemainingInBatch(
+  seconds: number,
+  callback: (SResult: SecondsRemainingResult) => void,
+): void {
+  // allow for fluid callback change without resetting interval
+  const callbackRef = useRef(callback)
+  callbackRef.current = callback
+
+  useEffect(() => {
+    // how often to check
+    // when there are ${seconds} seconds left in the batch
+    const checkIntervalSeconds = BATCH_TIME - seconds
```

Comment:
nit: you are effectively operating 2 constants, I understand the idea behind doing this "interval calculation" (you want to minimize the checks). I get why you do the check in line 43, but I think that console.warn should be a throw or at least a more noisy error, since this case would be clearly a misconfiguration